core/src/crypto/mod.rs

```diff
@@ -48,7 +53,11 @@ pub fn unblind_signature(
     Ok(Signature(res.compress().0))
 }
 
-pub fn verify(public_key: &PublicKey, message: &[u8], signature: Signature) -> Result<bool> {
+pub fn verify<P: Pair>(
```

Comment:
I think the verification functions should be generic over the rng by themselves, so we don't have to tie a signature scheme to a rng